packages/gatsby-plugin-typedoc/src/gatsby-node.ts

```diff
@@ -73,6 +77,37 @@ export function createSchemaCustomization({
         SomeValue: { value: 2097248 }
       }
     }),
+    schema.buildObjectType({
+      name: `TypedocMdx`,
+      extensions: ['dontInfer'],
+      fields: {
+        body: 'String!',
+        mdxAST: 'JSON'
+      }
+    }),
+    schema.buildObjectType({
+      name: `TypedocMarkdownRemark`,
+      extensions: ['dontInfer'],
+      fields: {
+        html: 'String'
+      }
+    }),
+    schema.buildObjectType({
+      name: 'TypedocCommentValue',
+      interfaces: ['Node'],
+      extensions: ['dontInfer'],
+      fields: {
+        value: 'String',
+        markdownRemark: {
+          type: 'TypedocMarkdownRemark',
+          resolve: proxyToNode('fields.markdownRemark')
```

Comment:
I think you are missing https://github.com/jquense/docpocalypse/blob/master/packages/theme-core/plugins/react-docgen/gatsby-node.js#L267 to make this work,

These fields are set by a `createNodeField` call. What is happening in the react-docgen case is this:

- create a description node
  - MDX sees a new `text/markdown` node and creates a MDX node, with that node as it's parent
- react-docgen plugin sees the new MDX node, check if it's `parent` is a Description
  - if so it tags the description with the mdx ID so it can resolve it easier a bit later. It's sort of like adding a backref

Then in the description resolver, the logic is 

- Grab node that corresponds to the ID stored at this path (fields.mdx)"
- Get every field matching field on my GQL type, that has a corresponding field on the other type
  - use the resolver for the other field as the resolver for my field of the same name

E.g`TypedocMDX.body` use the `MDX.body` field def to resolve the value

The reason this is convoluted is if you just link to the MDX type, it MUST exist in the graphql schema, b/c strong typing. Since MDX is optional, e.g. a user may not have it configured we need to make a proxy type, that way if MDX isn't configured our types still exist, but resolve to null